src/Integration/Telemetry/TelemetryManager.cs

```diff
@@ -213,10 +213,13 @@ public void LanguageAnalyzed(string languageKey)
             Debug.Assert(!string.IsNullOrEmpty(languageKey), "Supplied languageKey should not be null/empty");
             Debug.Assert(telemetryRepository.Data != null);
 
-            if (!telemetryRepository.Data.Analyses.Any(x => string.Equals(x.Language, languageKey, StringComparison.OrdinalIgnoreCase)))
+            lock (Lock)
```

Comment:
As we discussed offline, lack of Lock resulted in non thread-safe access to the list: it was possible for List.Any to execute while was another thread was calling List.Add.